File: cases/taint/python/return-relay-two-hop-positive/return_relay.py
def dfb_source():  # DFB-SOURCE: return-two-hop-input
    return "tainted"


def first_relay(value):  # DFB-WITNESS: return-two-hop-first
    return value


def second_relay(value):  # DFB-WITNESS: return-two-hop-second
    return first_relay(value)


def dfb_sink(value):  # DFB-SINK: return-two-hop-sink
    pass


def run():
    result = second_relay(dfb_source())
    dfb_sink(result)

File: cases/taint/ruby/return-relay-one-hop-positive/return_relay.rb
def dfb_source # DFB-SOURCE: return-one-hop-input
  "tainted"
end

def relay(value) # DFB-WITNESS: return-one-hop-relay
  value
end

def dfb_sink(value) # DFB-SINK: return-one-hop-sink
end

def run
  result = relay(dfb_source)
  dfb_sink(result)
end
